### PR TITLE
Debug build

### PR DESCRIPTION
**Pull request by warniaha**
Add defines for DAOS_NOT_IMPLEMENTED_LOG & DAOS_NOT_IMPLEMENTED_GDB_BREAK.  Logging includes the filename, line number & function name of the method.
Add method DebugBreak() to be able to break into the debugger from within the code

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/rgw/rgw_sal_daos.h [hunk 1]
```diff
@@ -34,11 +34,76 @@
 #include "rgw_role.h"
 #include "rgw_sal.h"
 
+inline bool IsDebuggerAttached()
```
Comment (zalsader) [inline src/rgw/rgw_sal_daos.h:37]: This is brilliant, thank you!
I think these could live in their own header to become available to more ceph devs

File: cmake/modules/FindDAOS.cmake [hunk 2]
```diff
@@ -32,6 +32,9 @@ set(DAOS_INCLUDE_DIRS ${DAOS_INCLUDE_DIR} ${DAOS_FS_INCLUDE_DIR})
 set(DAOS_LIBRARIES ${DAOS_LIBRARY} ${DAOS_FS_LIBRARY} ${DAOS_UNS_LIBRARY})
 
 include(FindPackageHandleStandardArgs)
+include_directories( ${PC_DAOS_INCLUDEDIR} )
+link_directories( ${PC_DAOS_LIBDIR} )
+
```
Comment (zalsader) [inline cmake/modules/FindDAOS.cmake:37]: Are these necessary here?

File: src/rgw/CMakeLists.txt [hunk 3]
```diff
@@ -272,7 +272,10 @@ endif()
 
 if(WITH_RADOSGW_DAOS)
   find_package(DAOS REQUIRED)
+  set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -DDEBUG ")
   target_link_libraries(rgw_common PRIVATE daos dfs uuid duns)
+  target_include_directories(rgw_common PRIVATE ${PC_DAOS_INCLUDEDIR} )
+  link_directories( ${PC_DAOS_LIBRARY_DIRS} )
```
Comment (zalsader) [inline src/rgw/CMakeLists.txt:278]: Those too

**Review by zalsader — APPROVED** [review]
Looks good, thank you for adding this

File: cmake/modules/FindDAOS.cmake (hunk 2 above)
Comment (warniaha) [reply cmake/modules/FindDAOS.cmake:37]: yes, for when building DAOS without forcing the folder to build in.  These are for locating the files

**Pull request merged**